param/__init__.py

```diff
@@ -85,14 +86,22 @@ def as_unicode(obj):
     return unicode(obj)
 
 
-def named_objs(objlist):
+def named_objs(objlist, namesdict=None):
     """
     Given a list of objects, returns a dictionary mapping from
-    string name for the object to the object itself.
+    string name for the object to the object itself. Accepts
+    an optional name,obj dictionary, which will override any other
+    name if that item is present in the dictionary.
     """
     objs = OrderedDict()
+
+    if namesdict is not None:
+        objtoname = {v: k for k, v in namesdict.items()}
```

Comment:
This is slightly confusing to me: this is "invert the dict"? Should I worry about a dictionary value being unsuitable for use as a key? (Or multiple keys pointing to the same object? Although that one seems a bit weird if I think of a selector in a gui, maybe it'd be ok in general?)

Reply:
Yes, this is inverting the dict, because it needs to find the name for a given object, but the function accepts a mapping from name to object, which is what the user would expect.

Yes, this will be limited to hashable objects; I'll update the docstring.  If the user supplies multiple keys pointing to the same object, I'd say that's their own problem. :-)